tools/pot/openvino/tools/pot/graph/passes.py

```diff
@@ -804,19 +857,30 @@ def insert_fake_quantize(graph, node, ports=None, names=None):
 
         # This condition blocks FQ insertion after the keep_shape_ops (KSO) generated sub-graph
         # to avoid quantization of integer-like tensors
-        if port.get_source().node.type != 'Const' and port.data.get_value() is not None:
+        if in_port_type != 'Const' and port.data.get_value() is not None:
             continue
 
-        name = 'fq_input'
+        is_weights = in_port_type == 'Const'
```

Comment:
Isn't `in_port_type` always `Const` here because we may not go here through `in_port_type != 'Const'`.
If so, `is_weight` must be always `True` for here.

Reply:
According to
L859: if in_port_type != 'Const' and port.data.get_value() is not None:
L860:        continue

In the case of 'in_port_type = 'Const', a program will proceed to L861,
In the case of 'in_port_type = 'MatMul' and 'port.data.get_value() = None', also will proceed.

So, we should take care about both 'in_port_type = Const' and 'in_port_type != Const' after L861.